packages/next/build/webpack/loaders/next-middleware-ssr-loader/index.ts

```diff
@@ -156,7 +105,14 @@ export default async function middlewareRSCLoader(this: any) {
             basePath: ${JSON.stringify(basePath || '')},
             supportsDynamicHTML: true,
             concurrentFeatures: true,
-            renderServerComponent: ${isServerComponent ? 'true' : 'false'},
+            renderServerComponentData,
+            renderServerComponent: ${
+              !isServerComponent
+                ? 'null'
+                : `
+              props => renderToReadableStream(createElement(Page, props), rscManifest)
+            `
+            },
```

Comment:
Conceptually, Server Component data should replace `getServerSideProps`. Is there a reason not to move this entirely into `render` and just pass e.g. `rscManifest` down?

Reply:
Nope passing the manifest is the only reason. Gonna refactor it!